src/adapters/http/sendNotification/router.ts

```
@@ -1,28 +1,36 @@
 import express from 'express';
-import * as f from 'fp-ts/function';
+import { pipe, flow } from 'fp-ts/lib/function';
 import * as E from 'fp-ts/Either';
-import * as utils from '../utils';
+import * as TE from 'fp-ts/TaskEither';
 import { ApiKey } from '../../../generated/definitions/ApiKey';
 import { NewNotificationRequest } from '../../../generated/definitions/NewNotificationRequest';
 import { SendNotificationUseCase } from '../../../useCases/SendNotificationUseCase';
+import { Handler, toExpressHandler } from '../Handler';
+import { makeProblem } from '../codec';
 
 const handler =
-  (sendNotificationUseCase: SendNotificationUseCase): express.Handler =>
+  (sendNotificationUseCase: SendNotificationUseCase): Handler =>
   (req, res) =>
-    f.pipe(
+    pipe(
       E.of(sendNotificationUseCase),
       E.ap(ApiKey.decode(req.headers['x-api-key'])),
       E.ap(NewNotificationRequest.decode(req.body)),
-      utils.sendResponse(res)(
-        (_error) => res.status(500).send(utils.makeAPIProblem(500, 'Internal Server Error')(undefined)),
-        (_) => res.status(202).send(_.returned)
+      // Create response
+      E.map(
+        flow(
+          TE.bimap(
+            (_) => res.status(500).send(makeProblem(500)),
```

Far better, thank you! 

A question: why are discarding errors?